src/plugins/dashboard/public/application/dashboard_app_controller.tsx

```diff
@@ -188,16 +187,24 @@ export class DashboardAppController {
     const stopSyncingAppFilters = connectToQueryState(
       queryService,
       {
-        set: ({ filters }) => dashboardStateManager.setFilters(filters || []),
-        get: () => ({ filters: dashboardStateManager.appState.filters }),
+        set: ({ filters, queryString }) => {
+          dashboardStateManager.setFilters(filters || []);
+          dashboardStateManager.setQuery(queryString || queryStringMananger.getDefaultQuery());
```

Comment:
How did this work before your change? Did the local dashboard state even update correctly on upstream changes?

Reply:
I think its something around `updateQueryAndFetch` - I think it can probably be deleted now

Reply:
In this  case it would be cool if you  could clean it up on this PR as well.

Reply:
@flash1293 You can see I just did that for `discover`.
Will try with dashboard but it might require some assistance @majagrubic